File: tests/integration/features/file_traceability/_language_parsers/python/02_python_function/file.py
def hello_world():
    """
    Some text.

    @relation(REQ-1, scope=function)
    """
    # @relation(REQ-1, scope=range_start)
    print("hello world")  # noqa: T201
    # @relation(REQ-1, scope=range_end)
    print("hello world")  # noqa: T201
    # @relation(REQ-1, scope=range_start)
    print("hello world")  # noqa: T201
    # @relation(REQ-1, scope=range_end)

def non_covered_function():
    pass

File: tests/integration/features/source_code_traceability/_source_nodes/11_functions_without_fields_do_not_generate_node/tests/file.c
#include <stdio.h>

/**
 * Some text.
 *
 * @relation(REQ-1, scope=function)
 *
 * INTENTION: Intention...
 *
 * INPUT: This
 *        is
 *        a statement
 *        \n\n
 *        And this is the same statement's another paragraph.
 *
 * EXPECTED_RESULTS: This is a comment.
 */
void test_case_1(void) {
    print("hello world\n");
}

/**
 * Plain documentation, no fields. Don't auto-generate a SDoc node for this function.
 *
 * @relation(REQ-2, scope=function)
 */
void test_case_2(void) {
    print("hello world\n");
}
